sql/core/src/main/scala/org/apache/spark/sql/parquet/newParquet.scala

```diff
@@ -630,11 +635,12 @@ object ParquetRelation2 {
             sqlContext.conf.isParquetBinaryAsString,
             sqlContext.conf.isParquetINT96AsTimestamp))
       }
-    }.reduce { (left, right) =>
-      try left.merge(right) catch { case e: Throwable =>
-        throw new SparkException(s"Failed to merge incompatible schemas $left and $right", e)
-      }
-    }
+    }.foldLeft[StructType](null) {
```

Comment:
Yeah, I was trying that also, but seems using `null` is more simple, as `Option` requires some more value extracting code.